From 14ea1f0f43c5f0324254ee41eb4af2316a8b975a Mon Sep 17 00:00:00 2001
From: Oliver Giles <ohw.giles@gmail.com>
Date: Sat, 9 Jan 2021 21:51:12 +1300
Subject: [PATCH] move ansi_up to latest version: 4.0.4

this is also the version in debian libjs-ansi-up
---
 CMakeLists.txt          | 4 ++--
 src/resources/js/app.js | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index eb17ee5..eddbcc5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,8 +85,8 @@ add_custom_command(OUTPUT index_html_size.h
 # Download 3rd-party frontend JS libs...
 file(DOWNLOAD https://cdnjs.cloudflare.com/ajax/libs/vue/2.6.12/vue.min.js
 	js/vue.min.js EXPECTED_MD5 fb192338844efe86ec759a40152fcb8e)
-file(DOWNLOAD https://raw.githubusercontent.com/drudru/ansi_up/v1.3.0/ansi_up.js
-        js/ansi_up.js EXPECTED_MD5 158566dc1ff8f2804de972f7e841e2f6)
+file(DOWNLOAD https://raw.githubusercontent.com/drudru/ansi_up/v4.0.4/ansi_up.js
+        js/ansi_up.js EXPECTED_MD5 b31968e1a8fed0fa82305e978161f7f5)
 file(DOWNLOAD https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.7.2/Chart.min.js
         js/Chart.min.js EXPECTED_MD5 f6c8efa65711e0cbbc99ba72997ecd0e)
 # ...and compile them
diff --git a/src/resources/js/app.js b/src/resources/js/app.js
index 3e7f165..3c1f5dd 100644
--- a/src/resources/js/app.js
+++ b/src/resources/js/app.js
@@ -564,6 +564,7 @@ const Job = templateId => {
 // Component for the /job/:name/:number endpoint
 const Run = templateId => {
   const utf8decoder = new TextDecoder('utf-8');
+  const ansi_up = new AnsiUp;
   const state = {
     job: { artifacts: [], upstream: {} },
     latestNum: null,
